src/App.tsx

```diff
@@ -1,14 +1,76 @@
 /* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
+import React, { useEffect, useState } from 'react';
+
+import { getTodos } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+
+import { Todo } from './types/Todo';
+import { FilterStatus } from './types/FilterStatus';
+
+import 'bulma/css/bulma.css';
+import '@fortawesome/fontawesome-free/css/all.css';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [filterStatus, setFilterStatus] = useState(FilterStatus.ALL);
+
+  useEffect(() => {
+    getTodos()
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        throw new Error('Loading todos error: ', error.message);
+      });
+  }, []);
+
+  const visibleTodos = todos.filter((todo) => {
+    const normalizedQuery = query.toLowerCase().trim();
```

Comment:
You are filtering 'todos' on every render. Consider using useMemo to optimize this operation and avoid unnecessary recalculations.